release/models/route-programming/openconfig-route-programming-state.yang

```diff
@@ -0,0 +1,199 @@
+          "Surrounding container for the list of routes that fail programming.";
+
+        list failed {
+          key "prefix";
+
+          description
+            "A prefix that was attempted to the programmed into hardware, but the
+            programming operation failed.";
+
+          leaf prefix {
+            type leafref {
+              path "../state/prefix";
+            }
+            description
+              "Reference to the prefix that keys the failed list.";
+          }
+
+          container state {
+            description
+              "Operational state parameters relating to a failed programming
+              operation.";
+            uses route-state-common;
+          }
+        }
+      }
+
+      container stale-routes {
+        description
+          "Surrounding container for the list of routes that are currently in
+          a stale state.";
+
+        list stale {
+          key "prefix";
+
+          description
+            "A prefix that is currently installed in hardware, but a subsequent
+            operation to update its programming failed - such that the entry in
+            hardware is stale.";
+
+          leaf prefix {
+            type leafref {
+              path "../state/prefix";
+            }
+            description
+              "Reference to the prefix that keys the stale list.";
+          }
+
+          container state {
+            description
+              "Operational state parameters relating to a stale route.";
+            uses route-state-common;
+          }
+        }
+      }
+
+      container drop-routes {
+        description
+          "Surrounding container for the list of routes that are currently in
+          a drop state.";
+
+        list drop {
+          key "prefix";
+
+          description
+            "A prefix that is currently installed in hardware but with an explicit
+            instruction that it should discard packets that are destined towards
+            it.";
+
+          leaf prefix {
+            type leafref {
+              path "../state/prefix";
+            }
+            description
+              "Reference to the prefix that keys the drop list.";
+          }
+
+          container state {
+            description
+              "Operational state parameters relating to a drop route.";
+            uses route-state-common;
+          }
+        }
+      }
+    }
+  }
+
+ grouping route-state-common {
+    description
+      "Common state parameters that correspond to a particular route
+      type.";
+
+    leaf prefix {
+      type oc-inet:ip-prefix;
+      description
+        "The IPv4 or IPv6 prefix that the route state corresponds to.";
+    }
+
+    leaf time {
+      type oc-types:timeticks64;
+      description
+        "Represents the time the route programming state change was detected by the 
+        monitoring subsystem expressed as nanoseconds since the Unix epoch.";
+    }
+
+    leaf dest-component {
```

Comment:
If `des-component` is for providing more error details for a modular chassis, should we consider:
1. make it a list? For example, multiple linecard could have programming failures. 
2. move the `reason` under each component, because each component might have different failure reasons? 

Another approach is
1. Remove `dest-component`.
2. Make `reason` a container where we can add a new filed to store string messages. The field can then be populated by devices to provide more info in terms of which component failed the programming? 

I feel like the alternative is more flexible, because we can always add back `dest-component` later if needed? 

cc: @earies